aiida_workgraph/engine/workgraph.py

```diff
@@ -646,9 +682,27 @@ def update_task_state(self, name: str) -> None:
             self.task_set_context(name)
             self.report(f"Task: {name} finished.")
         else:
-            task["results"] = None
+            task.setdefault("results", None)
+
+        self.update_parent_task_state(name)
 
+    def set_normal_task_results(self, name, results):
```

Comment:
Is normal task somewhere explained in the docstring? If not could you explain it somewhere? It becomes clear from reading the code, but this should be also somewhere documented

Reply:
Thanks! I added one line to the docstring of this function.
```
A  normal task is created by decorating a function with @task()
```